velox/vector/BaseVector.cpp

```diff
@@ -667,6 +667,47 @@ VectorPtr newConstant<TypeKind::OPAQUE>(
       pool, size, value.isNull(), type, std::shared_ptr<void>(capsule.obj));
 }
 
+template <TypeKind kind>
+VectorPtr newConstantFromString(
+    const TypePtr& type,
+    const std::optional<std::string>& value,
+    vector_size_t size,
+    velox::memory::MemoryPool* pool) {
+  using T = typename KindToFlatVector<kind>::WrapperType;
+  if (!value.has_value()) {
+    return std::make_shared<ConstantVector<T>>(pool, size, true, type, T());
+  }
+
+  if (type->isDate()) {
+    auto copy =
+        util::castFromDateString(StringView(value.value()), true /*isIso8601*/);
+    return std::make_shared<ConstantVector<int32_t>>(
+        pool, size, false, type, std::move(copy));
+  }
+
+  if constexpr (std::is_same_v<T, StringView>) {
+    return std::make_shared<ConstantVector<StringView>>(
+        pool, size, false, type, StringView(value.value()));
+  } else {
+    auto copy = velox::util::Converter<kind>::cast(value.value());
+    if constexpr (kind == TypeKind::TIMESTAMP) {
+      copy.toGMT(Timestamp::defaultTimezone());
+    }
+    return std::make_shared<ConstantVector<T>>(
+        pool, size, false, type, std::move(copy));
+  }
+}
+
+// static
+VectorPtr BaseVector::createConstant(
```

Comment:
I would keep this private in hive connector as this way of parsing string is very specific to the partition string value

Reply:
fixed.